python/kserve/kserve/model_server.py

```diff
@@ -212,71 +120,31 @@ def start(self, models: Union[List[Model], Dict[str, Deployment]]) -> None:
         else:
             raise RuntimeError("Unknown model collection types")
 
-        cfg = uvicorn.Config(
-            self.create_application(),
-            host="0.0.0.0",
-            port=self.http_port,
-            workers=self.workers,
-            log_config={
-                "version": 1,
-                "formatters": {
-                    "default": {
-                        "()": "uvicorn.logging.DefaultFormatter",
-                        "datefmt": DATE_FORMAT,
-                        "fmt": "%(asctime)s.%(msecs)03d %(name)s %(levelprefix)s %(message)s",
-                        "use_colors": None,
-                    },
-                    "access": {
-                        "()": "uvicorn.logging.AccessFormatter",
-                        "datefmt": DATE_FORMAT,
-                        "fmt": '%(asctime)s.%(msecs)03d %(name)s %(levelprefix)s %(client_addr)s %(process)s - '
-                               '"%(request_line)s" %(status_code)s',
-                        # noqa: E501
-                    },
-                },
-                "handlers": {
-                    "default": {
-                        "formatter": "default",
-                        "class": "logging.StreamHandler",
-                        "stream": "ext://sys.stderr",
-                    },
-                    "access": {
-                        "formatter": "access",
-                        "class": "logging.StreamHandler",
-                        "stream": "ext://sys.stdout",
-                    },
-                },
-                "loggers": {
-                    "uvicorn": {"handlers": ["default"], "level": "INFO"},
-                    "uvicorn.error": {"level": "INFO"},
-                    "uvicorn.access": {"handlers": ["access"], "level": "INFO", "propagate": False},
-                },
-            }
-        )
-
         if self.max_asyncio_workers is None:
             # formula as suggest in https://bugs.python.org/issue35279
-            self.max_asyncio_workers = min(32, utils.cpu_count()+4)
+            self.max_asyncio_workers = min(32, utils.cpu_count() + 4)
         logging.info(f"Setting max asyncio worker threads as {self.max_asyncio_workers}")
         asyncio.get_event_loop().set_default_executor(
             concurrent.futures.ThreadPoolExecutor(max_workers=self.max_asyncio_workers))
 
         async def serve():
             serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             serversocket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            serversocket.bind((cfg.host, cfg.port))
+            serversocket.bind(('0.0.0.0', self.http_port))
```

Comment:
is it ok to convert this to being hard coded?

Reply:
not sure if this was for testing so just want to check

Reply:
the host is not really configurable, by default it binds to localhost which is only accessible to process running on the same node, `0.0.0.0` tells to be reachable from a public IP of the container. Since you don’t know which ip address the container will be assigned within the subnetwork, 0.0.0.0 says to fastapi to accept incoming connections no matter the network adapter or the ip address.